regression/goto-instrument/restrict-function-pointer-to-complex-expression/test.c

```diff
@@ -0,0 +1,59 @@
+#include <assert.h>
+
+typedef int (*fptr_t)(int);
+
+fptr_t get_f(void);
+fptr_t get_g(void);
+
+void use_fg(int choice, fptr_t fptr, fptr_t gptr)
+{
+  assert((choice ? fptr : gptr)(10) == 10 + choice);
+}
+
+// this is just here to distinguish the behavior from FP removal, which'd include g
```

Comment:
:question: I don't understand this comment, or this variable

Reply:
Argh, I swear I had a version of this that fixed this. The reason this is here is to trick function pointer removal basically.